lib/matplotlib/widgets.py

```diff
@@ -973,45 +1048,259 @@ def clear(self, event):
+        if self.__pressed:
+            if self.ignore(event):
+                return
+            if not self.canvas.widgetlock.available(self):
+                return
+            if event.inaxes != self.ax:
+                return
+
+            # Store the new coordinates
+            self._coord[0] = float(event.xdata)
+            self._coord[1] = float(event.ydata)
+
+            self._draw_cursor()
+
+class Cursors(AxesWidget):
+    """
+    Several Cursors: horizontal and vertical lines that span the axes and
+    move with the pointer.
+
+    For the cursors to remain responsive you must keep a reference to them.
+
+    The following attributes are accessible
+
+      *ax*
+        The :class:`matplotlib.axes.Axes` the cursor renders in.
+
+      *properties*
+        a list of dictionaries for each cursor, containing
+
+          *horizOn*
+            Controls the visibility of the horizontal line
+
+          *vertOn*
+            Controls the visibility of the vertical line
+
+          *visible*
+            Controls the visibility of the cursor itself
+
+    The following code will add three cursors to an existing axis
+
+    csr = Cursors(ax, [{'color':'red'}, {'color':'blue'}, {'color':'black'}],
+              useblit=False)
+    """
+
+    def __init__(self, ax, properties, useblit=False):
+        """
+        Parameters
+        ----------
+        ax : matplotlib.axes.Axes
+            The :class:`matplotlib.axes.Axes` instance the cursor
+            will be placed into.
+
+        properties: list
+            list of dictionaries, one for each each cursor defining the
+            cursor's properties with keys:
+
+            horizOn : boolean
+                Controls the visibility of the horizontal line.
+
+            vertOn : boolean
+                Controls the visibility of the vertical line.
+
+            and all allowed line property keywords
+
+
+        useblit : boolean
+            use the backend-dependent blitting features for faster
+            updates (GTKAgg only for now).
+
+        .. plot :: mpl_examples/widgets/cursor.py
+        """
+        #TODO: add to setup dictionary
+        # coord : np.array
+        # Initial location of cursor
+
+
+        # TODO: Is the GTKAgg limitation still true?
+        super().__init__(ax)
+
+        self.connect_event('motion_notify_event', self.on_move)
+
+        self.connect_event('button_press_event', self.on_press)
+        self.connect_event('button_release_event', self.on_release)
+        self.__selected = None
+
+        self.connect_event('draw_event', self.clear)
+
+        self.useblit = useblit and self.canvas.supports_blit
+
+        self.background = None
+        self.needclear = False
+
+        # Initilize empty list
+        self.properties = [{} for prop in properties]
+
+        for prop, new_prop in zip(properties, self.properties):
+            new_prop['visible'] = True
+            new_prop['horizOn'] = prop.pop('horizOn', True)
```

Comment:
If you are going to pop, make a copy first.  Keeping functions pure is a general a good thing.  The use case this supports is passing the same list of properties to several `Cursors` each on a different axes.